app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -291,14 +291,22 @@ def create_vm(args = {})
       vm
     end
 
-    def preallocate_disks(args)
-      change_allocation_volumes = args[:volumes_attributes].values.select { |x| x[:id].present? && x[:preallocate] == '1' }
-      if args[:template].present? && change_allocation_volumes.present?
-        disks = change_allocation_volumes.map do |volume|
-          { :id => volume[:id], :sparse => 'false', :format => 'raw', :storagedomain => volume[:storage_domain] }
+    def preallocate_and_clone_disks(args, template)
+      volumes_to_change = args[:volumes_attributes].values.select {|x| x[:id].present?}
+      return unless volumes_to_change.present?
+
+      template_disks = template.volumes
+
+      disks = []
+      volumes_to_change.each do |volume|
```

Comment:
@ShimShtein I changed the `map` to `each` to avoid the nil value  if the `if` on line 306 if false.

`fog-ovirt` still needs to be fixed but it shouldn't block testing.

Reply:
IMHO `map.compact` looks a bit better.